tests/testament/htmlgen.nim

```diff
@@ -90,77 +70,88 @@ type
     totalCount, successCount, ignoredCount, failedCount: int
     successPercentage, ignoredPercentage, failedPercentage: BiggestFloat
 
-proc allTestResults(): AllTests =
+proc allTestResults(onlyFailing = false): AllTests =
   result.data = newJArray()
-  for file in os.walkFiles("testresults/*.json"):
-    let data = parseFile(file)
+  for file in os.walkFiles("testresults"/"*.json"):
+    var data: JsonNode
+    try: data = parseFile(file)
+    except JsonParsingError:
+      echo "[ERROR] JSON parsing error: ", getCurrentExceptionMsg()
```

Comment:
This is an additional improvement, but it would be nice to use the logging module for this.

Reply:
My opinion on this matter is: Logging is always bad, don't do it.

Reply:
I think using logging is too much for just this **one** instance of us printing here... In a larger system, I could be persuaded to do it with logging... But as I pointed out earlier this is just for one very particular special case.

Reply:
It doesn't have to be logging, but at least provide a proc called ``reportError`` or something. So that we can easily modify the output format.